### ISSUE TITLE
Failed to read Git log: fatal: detected dubious ownership in repository at '/github/workspace'

### ISSUE THREAD

**Issue by jpopelka** [benmatselby/hugo-deploy-gh-pages#49]
Hello,

the action started to fail recently for one of our repos.
```
Building the Hugo site with: 'hugo '
Start building sites … 
hugo v0.98.0-...
ERROR 2022/12/20 10:14:35 Failed to read Git log: fatal: detected dubious ownership in repository at '/github/workspace'
To add an exception for this directory, call:
	git config --global --add safe.directory /github/workspace
Error: Error building site: logged 1 error(s)
```

I tried to reproduce it locally by checking out the code the way GitHub does and then running your action in a container as described in the README, but I haven't been successful so I don't have a clear reproducer for you.
But if you fork https://github.com/packit/status the https://github.com/packit/status/blob/main/.github/workflows/deploy-pages.yml should fail for you as well - if you change it to use your action instead of my fork and remove the `if:` condition.

I have been able to workaround the problem exactly the way the error message suggests, i.e. by [adding
"git config --global --add safe.directory ${PWD}" to the `action.sh`](https://github.com/jpopelka/hugo-deploy-gh-pages/commit/df7c2381c2e8837a0822007a08c808fd6661f152).

See actions/checkout#766 for more info.

**Comment by benmatselby** [comment]
Hey there @jpopelka 

Just a note to say I've seen this, but currently cannot replicate this. This is the config I've just deployed with

```
      - name: Deploy the site
        uses: benmatselby/hugo-deploy-gh-pages@main
        env:
          TARGET_REPO: benmatselby/benmatselby.github.io
          TARGET_BRANCH: main
          TOKEN: ${{ secrets.TOKEN }}
          CNAME: benmatselby.dev
```

**Comment by dashaun** [comment]
I think this means that your TOKEN is expired.

**Comment by dashaun** [comment]
@jpopelka You should submit that change as a PR

I've just started seeing this recently as well.

I found this on StackOverflow:
https://stackoverflow.com/questions/72978485/git-submodule-update-failed-with-fatal-detected-dubious-ownership-in-repositor

My repo has submodules.

**Comment by brunoaugusto** [comment]
How can I *actually* fix this?

I've tried several "solutions" and workarounds involving `git config` (even though I don't have the directory mentioned in the error among my files), tried to [take ownership](https://confluence.atlassian.com/bbkb/git-command-returns-fatal-error-about-the-repository-being-owned-by-someone-else-1167744132.html) of the directory and nothing worked. In the end, I had to do an update like in the old days, cloning my public repository, compiling Hugo files and committing manually.

Everything was running just fine the past month when I did the last update before the holidays.

**Comment by jpopelka** [comment]
> You should submit that change as a PR

@dashaun #50 

> How can I _actually_ fix this?

@brunoaugusto The only way I found is #50. If you're still having the issue, you can try my fork@branch
```
uses: jpopelka/hugo-deploy-gh-pages@safe-directory
```
until it's fixed in this repo.

**Comment by benmatselby** [comment]
Thanks to @jpopelka for the fix. I've just released [v2.1.0](https://github.com/benmatselby/hugo-deploy-gh-pages/releases/tag/v2.1.0) of this action.
